Sources/SwiftParser/Recovery.swift

```diff
@@ -123,13 +123,29 @@ extension Parser.Lookahead {
       if currentTokenPrecedence >= recoveryPrecedence {
         break
       }
-      self.consumeAnyToken()
       if let closingDelimiter = currentTokenPrecedence.closingTokenKind {
         let closingDelimiterSpec = TokenSpec(closingDelimiter)
+        let canSkipSingleAtSameLine = self.withLookahead { lookahead in
+          while !lookahead.at(.endOfFile), !lookahead.currentToken.isAtStartOfLine {
+            if lookahead.at(closingDelimiterSpec) {
+              return true
+            } else {
+              lookahead.consumeAnyToken()
+            }
+          }
+          return false
+        }
+        if canSkipSingleAtSameLine {
+          skipSingle()
+          continue
+        }
+        self.consumeAnyToken()
         guard self.canRecoverTo(closingDelimiterSpec) != nil else {
-          break
+          continue
```

Comment:
Yes, I think this makes a lot of sense so we don’t completely abort recovery on a mismatched `(`. Good catch!